agent/config/config_test.go

```diff
@@ -143,6 +158,9 @@ func TestGet(t *testing.T) {
 		}, &actual, logrus.WithField("test", t.Name()))
 		require.NoError(t, err)
 
+		tmpDir := generateTempDir(t)
+		defer removeTempDir(t, tmpDir)
```

Comment:
seems like we can move all these calls to the top level of these tests in TestGet

Reply:
Done.